### PR TITLE
MULE-16207: Make cacheID hash generator to only take into account parameters required for metadata

### PR DESCRIPTION
**Pull request by ndinu**

### CODE REVIEW

**Comment by ndinu** [comment]
--validate

**Comment by afelisatti** [comment]
--validate

File: modules/spring-config/src/main/java/org/mule/runtime/config/api/dsl/model/metadata/DslElementBasedMetadataCacheIdGenerator.java [hunk 1]
```diff
@@ -158,9 +162,12 @@ private String getSourceElementName(DslElementModel<?> elementModel) {
 
   private Optional<MetadataCacheId> resolveGlobalElement(DslElementModel<?> elementModel) {
     List<MetadataCacheId> parts = new ArrayList<>();
+    List<String> parameterNamesRequiredForMetadataCacheId =
+        parameterNamesRequiredForMetadataCacheId(elementModel.getModel()).orElse(null);
```
Comment (alegmarra) [inline modules/spring-config/src/main/java/org/mule/runtime/config/api/dsl/model/metadata/DslElementBasedMetadataCacheIdGenerator.java]: you can create the predicate inline here to do the filtering and avoid the null list ( ❌)

File: modules/spring-config/src/main/java/org/mule/runtime/config/api/dsl/model/metadata/DslElementBasedMetadataCacheIdGenerator.java [hunk 2]
```diff
@@ -173,7 +180,26 @@ private String getSourceElementName(DslElementModel<?> elementModel) {
...
+  private boolean isRequiredForMetadata(List<String> parameterNamesRequiredForMetadataCacheId, Object model) {
+    if (model instanceof ParameterModel && parameterNamesRequiredForMetadataCacheId != null) {
+      return parameterNamesRequiredForMetadataCacheId.contains(((ParameterModel) model).getName());
+    } else {
+      return true;
+    }
+  }
+
+  private Optional<List<String>> parameterNamesRequiredForMetadataCacheId(Object model) {
+    if (model != null && model instanceof EnrichableModel) {
```
Comment (alegmarra) [inline modules/spring-config/src/main/java/org/mule/runtime/config/api/dsl/model/metadata/DslElementBasedMetadataCacheIdGenerator.java]: instanceof is already nullsafe

File: modules/spring-config/src/main/java/org/mule/runtime/config/api/dsl/model/metadata/DslElementBasedMetadataCacheIdGenerator.java [hunk 3]
```diff
@@ -173,7 +180,26 @@ private String getSourceElementName(DslElementModel<?> elementModel) {
...
+      Optional<RequiredForMetadataModelProperty> modelProperty =
+          ((EnrichableModel) model).getModelProperty(RequiredForMetadataModelProperty.class);
+      if (modelProperty.isPresent()) {
```
Comment (alegmarra) [inline modules/spring-config/src/main/java/org/mule/runtime/config/api/dsl/model/metadata/DslElementBasedMetadataCacheIdGenerator.java]: return requiredParametersModelProperty.map(mp -> mp.getRequiredParameters())

**Pull request merged**